### ISSUE TITLE
[FR] Prefer alias for "absolute" (kind of) path in Node env

### ISSUE THREAD

**Issue by hoangnhan2ka3** [Limegrass/eslint-plugin-import-alias#16]
Hi, thank you for your efforts.

This is almost the only package that satisfies half of my problems regarding `alias import`.

First problem:

```tsx
// .eslintrc.js
rules: {
  "@limegrass/import-alias/import-alias": [1, {
    aliasConfigPath: "./tsconfig.json",
    relativeImportOverrides: [
      {
        path: ".",
        depth: 0
      }
    ]
  }]
}
```

Just with that base config, it solved:

```tsx
// from this
import { siteConfig } from "../config/site.config"
// to this
import { siteConfig } from "@/config/site.config"

// and also keep the path `relative` if the file was imported is in the same folder (depth: 0)
import Content from "./_content"
// nothing change
```

That was what I expected! 🚀

But may I ask if the second problem can be solved with this package?

That is force/ prefer alias also for some specific (?) folder that was set in config; which was `absolute` imported. Specifically here is the `src` folder (Sorry for the confusing words, I don't know how to put it into words for you to understand. I hope you can understand better through the code below).

Here's what I want:

```tsx
// from this
import { MotionArticle, MotionP } from "src/components/mdx/MotionText"
// to this
import { MotionArticle, MotionP } from "@/components/mdx/MotionText"
```

And here's my `tsconfig.json`:

```json
"compilerOptions": {
  //...
  "baseUrl": ".",
  "paths": {
      "#/*": [
          "./*",
      ],
      "@/*": [
          "./src/*",
          "./public/*"
      ],
      "@/app/(site)/*": [
          "./src/app/(site)/(footer)/*",
          "./src/app/(site)/(subpages)/*"
      ],
      "contentlayer/generated": [
          "./.contentlayer/generated"
      ],
      "data": ["./data.ts"]
  }
  //...
},
```

In short, I want when I import a `component` from the `src` folder `absolute` way, this package can help me automatically fix it so that it respects the alias I have set in `tsconfig.json`, in this means `from "src/" to "@/"`.

Thank you, have a nice day!

**Comment by Limegrass** [comment]
Hey, thanks for the question!

I'm not certain I understand your use case correctly, is what you were looking for to change `import Content from "./_content"`
to `import Content from @/content`? If so, that can be done through not including a `relativeImportOverrides` -- e.g.
```
rules: {
  "@limegrass/import-alias/import-alias": [1, {
    aliasConfigPath: "./tsconfig.json"
  }]
}
```

Otheriwse, in the case of `"src/components/mdx/MotionText"`, if `src` is still relative to the current file, currently it is still considered a relative import, just one that reaches into more nested directories. At the moment, it isn't currently supported to limit the... "branch depth" to just within the same folder. Is this the use case you were looking for?

If neither of those describe your issue properly, would you be able to explain your issue with an example project or a full description of the project layout, tsconfig, and attempted import?

The second use case was something I had considered initially; however, I took a shortcut to implementation for relative import override as I did intend to do a major rework of the plugin as a whole to support more granularity in general.
Let me know, and I can try to prioritize that accordingly!

**Comment by hoangnhan2ka3** [comment]
Thanks for your time!

> I'm not certain I understand your use case correctly, is what you were looking for to change import Content from "./_content"
> to import Content from @/content? If so, that can be done through not including a relativeImportOverrides

No, keep it to not change to `@/content` is my expectation (I mean keep it as `./_content`), cuz I just want to import some components `relative way` if they are in a same folder only. So this is not a problem.

Here's my simple folder structure, I'm using Nextjs `app directory` with `src` folder at the root level:

```
<root>/
├─ node_modules/
├─ src/
│  ├─ app/
│  │  ├─ layout.tsx
│  │  ├─ page.tsx
│  │  ├─ projects/
│  │  │  ├─ page.tsx <--(2)
│  │  │  ├─ _content.tsx
│  ├─ components/
│  │  ├─ MyComponent.tsx <--(1)
├─ eslint.config.js / .eslintrc.js
├─ tsconfig.json
├─ package.json

```

Let's say I want to import `MyComponent.tsx` (1) to `page.tsx` (2), there are almost three ways:

```tsx
// page.tsx

import MyComponent from "src/components/MyComponent"
// Absolute import from root
```
or
```tsx
// page.tsx

import MyComponent from "../components/MyComponent"
// Relative Import
```
or
```tsx
// page.tsx

import MyComponent from "@/components/MyComponent"
// In case u have set the alias `@/*` === `./src/*` in tsconfig.json
```

Both is valid, but I want every import to be _**consistent**_ by using `alias import` based on `compilerOptions.paths` in `tsconfig.json`:

> And here's my `tsconfig.json`:
> 
> ```json
> "compilerOptions": {
>   //...
>   "baseUrl": ".",
>   "paths": {
>       "#/*": [
>           "./*",
>       ],
>       "@/*": [
>           "./src/*",
>           "./public/*"
>       ],
>       "@/app/(site)/*": [
>           "./src/app/(site)/(footer)/*",
>           "./src/app/(site)/(subpages)/*"
>       ],
>       "contentlayer/generated": [
>           "./.contentlayer/generated"
>       ],
>       "data": ["./data.ts"]
>   }
>   //...
> },
> ```

I mean this:

```json
"@/*": [
   "./src/*"
],
```

So I hope the rule can help me fix `"src/components/MyComponent"` to `"@/components/MyComponent"`.

And hope you understand too, have another nice day!

**Comment by hoangnhan2ka3** [comment]
I think it should be a separate rule called `absoluteImportOverrides` or change the current rule's name from `relativeImportOverrides` to `aliasImportOverrides` if you can help me with my problem 🚀

**Comment by Limegrass** [comment]
Ah, thanks for the great explanation! I was actually unaware this was something supported by TypeScript, but this is certainly something we should support within the plugin. I'll take a look into potentially adding an option here.

**Comment by hoangnhan2ka3** [comment]
Thank you! Waiting for it

**Comment by Limegrass** [comment]
Hey @hoangnhan2ka3,

Sorry for the delay, I finally got around to work on this. I published it in [version 1.4.0](https://github.com/Limegrass/eslint-plugin-import-alias/releases/tag/v1.4.0), you'll need to include set a new option `isAllowBaseUrlResolvedImport: false` for your config for the rule unless you already [extend the recommended config](https://github.com/Limegrass/eslint-plugin-import-alias/blob/v1.3.2/examples/basic/.eslintrc.js#L6). The default value preserves the existing behavior of allowing the absolute import.

Feel free to re-open if that didn't actually solve this issue, and let me know if you face any problems. The implementation is definitely not the cleanest thing, so there's a fair chance there's a bug in there!

**Comment by hoangnhan2ka3** [comment]
Hi @Limegrass 

Do u mean I should set rules like this?

![image](https://github.com/Limegrass/eslint-plugin-import-alias/assets/147973044/8b56c461-13de-4ab2-90ea-326ab64b106f)

**Comment by hoangnhan2ka3** [comment]
One problem that when I set it in `rules`, nothing happen, no warning appear.

<img width="646" alt="image" src="https://github.com/Limegrass/eslint-plugin-import-alias/assets/147973044/e04d5e9c-9c97-437a-8dbf-f20c7b339367">

But when I use it in `extends` like this:

![image](https://github.com/Limegrass/eslint-plugin-import-alias/assets/147973044/73f31ef6-f075-4450-b873-aa938b436204)

It just works!

<img width="644" alt="image" src="https://github.com/Limegrass/eslint-plugin-import-alias/assets/147973044/902868f8-ab4b-40f3-992e-f054234c7e82">

Any bugs?

**Comment by hoangnhan2ka3** [comment]
Ahh I got about ~70 ESlint rules, I can't check it one by one :))

But hey I've found the problem, that when I set rules like [enabling-relative-import-overrides](https://github.com/Limegrass/eslint-plugin-import-alias/blob/main/docs/rules/import-alias.md#enabling-relative-import-overrides) and `isAllowBaseUrlResolvedImport: false` as u told me, it not working.

Work ✅

```js
"@limegrass/import-alias/import-alias": [1,
  {
    aliasConfigPath: "./tsconfig.json",
    isAllowBaseUrlResolvedImport: false
  }
]
```

Not work ❌

```js
"@limegrass/import-alias/import-alias": [1,
  {
    aliasConfigPath: "./tsconfig.json",
    relativeImportOverrides: [{
      path: ".",
      depth: 0
    }],
    isAllowBaseUrlResolvedImport: false
  }
]
```

I think the new rule is conflix with `relativeImportOverrides` option

**Comment by Limegrass** [comment]
@hoangnhan2ka3

Looks like you're right, thanks! This should be fixed in v1.4.1

**Comment by hoangnhan2ka3** [comment]
Yeah now it works like a charm

<img width="643" alt="image" src="https://github.com/Limegrass/eslint-plugin-import-alias/assets/147973044/5c68d48d-c1f9-46c2-9f38-bc3abbcaf5d1">

Thank you so much for your work @Limegrass 🚀

**Comment by Limegrass** [comment]
Thanks for the heads up, will take a look into adding some documentation there soon.
